pkg/platform/kube/test/platform_test.go

```diff
@@ -763,6 +764,93 @@ func (suite *DeployFunctionTestSuite) TestCreateFunctionWithTemplatedIngress() {
 		})
 }
 
+func (suite *DeployFunctionTestSuite) TestFunctionSecretCreation() {
+
+	// set platform config to support scrubbing
+	suite.PlatformConfiguration.SensitiveFields.MaskSensitiveFields = true
+
+	// reset platform configuration when done
+	defer func() {
+		suite.PlatformConfiguration.SensitiveFields.MaskSensitiveFields = false
+	}()
+
+	functionName := "func-with-secret"
+	password := "1234"
+	createFunctionOptions := suite.CompileCreateFunctionOptions(functionName)
+	secretName := fmt.Sprintf("%s%s", functionconfig.NuclioSecretNamePrefix, functionName)
+
+	createFunctionOptions.FunctionConfig.Spec.Build.CodeEntryAttributes = map[string]interface{}{
+		"password": password,
+	}
+
+	// delete function secrets when done
+	defer func() {
+		secrets, err := suite.KubeClientSet.CoreV1().Secrets(suite.Namespace).List(suite.Ctx, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", common.NuclioResourceLabelKeyFunctionName, functionName),
+		})
+		if err != nil {
+			suite.Logger.WarnWith("Failed to list secrets", "err", err)
+		}
+
+		for _, secret := range secrets.Items {
+			err := suite.KubeClientSet.CoreV1().Secrets(suite.Namespace).Delete(suite.Ctx, secret.Name, metav1.DeleteOptions{})
+			if err != nil {
+				suite.Logger.WarnWith("Failed to delete secret", "err", err, "secretName", secret.Name)
+			}
+		}
+	}()
```

Comment:
why does the test case needs to take care of it? once `DeployFunction` callback is done, it should delete the function. deleting function should take care of deleting the secrets as well.
meaning, the controller, upon delete function, will delete its secrets as well.

Reply:
The controller does delete the secrets, however it is a part of the next PR..
I will add a comment to remove this workaround in the next PR.